C error - multiple returns values and fixed arrays

### Describe the bug

When returning multiple values, if one of those is a fixed size array, it seems to be the cause of a C error.

### Reproduction Steps

```v
fn double() (bool, [4]i16) {
        return true, [4]i16{}
}

fn foo() {
        _, b := double()
        for i in b {
        }
}
``` 
with `v bug.v`

### Expected Behavior

No C error.

### Current Behavior

```
❯ v bug.v
bug.v:7:6: warning: unused variable: `i`
    5 | fn foo() {
    6 |     _, b := double()
    7 |     for i in b {
      |         ^
    8 |     }
    9 | }
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/bug.01J5EG61QWPZ7RVQCYVP6FH9BK.tmp.c:13289: error: invalid operand types for binary operation
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.7 294f7e4.8210593

### Environment details (OS name and version, etc.)

`OS: linux, Linux version 6.7.11-200.fc39.x86_64 (mockbuild@bdb8d24f206645e2bda44c1cf867612d) (gcc (GCC) 13.2.1 20240316 (Red Hat 13.2.1-7), GNU ld version 2.40-14.fc39) #1 SMP PREEMPT_DYNAMIC Wed Mar 27 16:50:39 UTC 2024`
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -5784,7 +5784,13 @@ fn (mut g Gen) return_stmt(node ast.Return) {
 						} else {
 							node.types[0]
 						}
-						g.write('memcpy(${tmpvar}.ret_arr, ${g.expr_string(node.exprs[0])}, sizeof(${g.typ(typ)}))')
+						typ_sym := g.table.final_sym(typ)
+						if typ_sym.kind == .array_fixed
+							&& (typ_sym.info as ast.ArrayFixed).is_fn_ret {
+							g.write('memcpy(${tmpvar}.ret_arr, ${g.expr_string(node.exprs[0])}.ret_arr, sizeof(${g.typ(typ)}))')
+						} else {
+							g.write('memcpy(${tmpvar}.ret_arr, ${g.expr_string(node.exprs[0])}, sizeof(${g.typ(typ)}))')
+						}
 					} else if node.exprs[0] in [ast.ArrayInit, ast.StructInit] {
 						if node.exprs[0] is ast.ArrayInit && node.exprs[0].is_fixed
 							&& node.exprs[0].has_init {

--- a/vlib/v/gen/c/for.v
+++ b/vlib/v/gen/c/for.v
@@ -333,6 +333,9 @@ fn (mut g Gen) for_in_stmt(node_ ast.ForInStmt) {
 				} else {
 					g.write(' = ${addr}')
 					g.expr(node.cond)
+					if info.is_fn_ret {
+						g.write('.ret_arr')
+					}
 					g.writeln('[${idx}];')
 				}
 			}

--- a/vlib/v/tests/array_fixed_for_loop_test.v
+++ b/vlib/v/tests/array_fixed_for_loop_test.v
@@ -1,0 +1,16 @@
+fn double() (bool, [4]i16) {
+	return true, [4]i16{}
+}
+
+fn foo() [4]i16 {
+	_, b := double()
+	for i in b {
+		println(i)
+	}
+	return b
+}
+
+fn test_main() {
+	arr := foo()
+	assert arr == [4]i16{}
+}

